callbacks/segmentation.py

```diff
@@ -1,50 +1,147 @@
+                "kwargs": {
+                    "job_type": "train",
+                    "experiment_id": "123",
+                    "dataset": "name_of_dataset",
+                    "params": '{"n_estimators": 30, "oob_score": true, "max_depth": 8}',
+                },
+            },
+        },
+        {
+            "mlex_app": "high-res-segmentation",
+            "description": "test_1",
+            "service_type": "backend",
+            "working_directory": "/data/mlex_repo/mlex_tiled/data",
+            "job_kwargs": {
+                "uri": "mlexchange1/random-forest-dc:1.1",
+                "type": "docker",
+                "cmd": "python segment.py data/data/20221222_085501_looking_from_above_spiralUP_CounterClockwise_endPointAtDoor_0-1000 data/seg-results-test/spiral/model/random-forest.model data/seg-results-test/spiral/output '{\"show_progress\": 1}'",
+                "kwargs": {
+                    "job_type": "train",
+                    "experiment_id": "124",
+                    "dataset": "name_of_dataset",
+                    "params": '{"show_progress": 1}',
+                },
+            },
+        },
+    ],
+    "host_list": ["vaughan.als.lbl.gov"],
+    "dependencies": {"0": [], "1": [0]},
+    "requirements": {"num_processors": 2, "num_gpus": 0, "num_nodes": 1},
+}
 
 
-# NEXT STEPS:
-# - this function returns a job ID, which would be associated with the workflow run on vaughan
-# - then we need another callback to pick up this ID and start polling for successful output
 @callback(
-    Output("output-placeholder", "children"),
+    Output("output-details", "children"),
+    Output("submitted-job-id", "data"),
     Input("run-model", "n_clicks"),
     State("annotation-store", "data"),
     State("project-name-src", "value"),
 )
 def run_job(n_clicks, annotation_store, project_name):
-    # As a placeholder, pulling together the inputs we'd need if we were going to submit a job
+    """
+    This callback collects parameters from the UI and submits a job to the computing api.
+    If the app is run from "dev" mode, then only a placeholder job_uid will be created.
+    The job_uid is saved in a dcc.Store for reference by the check_job callback below.
+
+    # TODO: Appropriately paramaterize the DEMO_WORKFLOW json depending on user inputs
+    and relevant file paths
+    """
     if n_clicks:
+        if MODE == "dev":
+            job_uid = str(uuid.uuid4())
+            return (
+                dmc.Text(
+                    f"Workflow has been succesfully submitted with uid: {job_uid}",
+                    size="sm",
+                ),
+                job_uid,
+            )
+        else:
+            data_utils.save_annotations_data(annotation_store, project_name)
```

Comment:
Testing on Vaughan gives us the following error:

```
File "/usr/local/lib/python3.9/site-packages/dash/_callback.py", line 450, in add_context
 output_value = func(*func_args, **func_kwargs)  # %% callback invoked %%
File "/app/callbacks/segmentation.py", line 83, in run_job
 data_utils.save_annotations_data(annotation_store, project_name)
File "/app/utils/data_utils.py", line 98, in save_annotations_data
 annotations.create_annotation_metadata()
File "/app/utils/annotations.py", line 61, in create_annotation_metadata
 self.set_annotation_image_shape(image_idx)
File "/app/utils/annotations.py", line 165, in set_annotation_image_shape
 self.annotation_image_shape = self.annotation_store["image_shapes"][0]
 KeyError: 'image_shapes'
```

I suspect this may come from interacting with a Tiled server that has only a single tiff-sequence in it, so we technically never actively selected a project. Interacting with the GUI more (changing slider value, 'selecting' the single project) does remove this error and the attempt to submit the job is made.

Reply:
@Wiebke I think this is happening because of [this line](https://github.com/mlexchange/mlex_highres_segmentation/blob/834b533639868dc318746dda45816186971ddd71/components/control_bar.py#L73), where my guess is that `DATA_OPTIONS` evaluates to `None`, which means that the slider is disabled so [this block isn't hit](https://github.com/mlexchange/mlex_highres_segmentation/blob/834b533639868dc318746dda45816186971ddd71/callbacks/image_viewer.py#L311-L314). 

I think you're probably right in that this because of a different structure on the Tiled server on your end. What's the structure of the `data` variable you get after running: 

```
client = from_uri(TILED_URI, api_key=API_KEY)
data = client["data"]
```